File: client/src/app/_models/homework.ts
import { SchoolSubject } from "./schoolSubject"

export interface Homework {
    id: number
    description: string
    comment: string
    teacherId: number
    teacherFullName?: string
    classId: number
    publishDate: Date 
    deadline: Date
    isActive: boolean
    subject: SchoolSubject
}

File: API/Entities/Homework.cs
using System;

namespace API.Entities;
public class Homework
{
    public int Id { get; set; }
    public string Description { get; set; }
    public string Comment { get; set; }
    public int TeacherId { get; set; }
    public int ClassId { get; set; }
    public DateTime PublishDate { get; set; } = DateTime.UtcNow;
    public DateTime Deadline { get; set; }
    public bool IsActive { get; set; } = true;
    public Subject Subject { get; set; }
}